legacy/c3_trajectory_generator/src/node.cpp

```diff
@@ -191,12 +242,35 @@ struct Node {
       c3trajectory_t = now;
     }
 
-    while (c3trajectory_t + traj_dt < now) {
-      c3trajectory->update(traj_dt.toSec(), current_waypoint,
-                           (c3trajectory_t - current_waypoint_t).toSec());
+    // Remember the previous trajectory
+    auto old_trajectory = c3trajectory->getCurrentPoint();
```

Comment:
Can you explain this? Does it just let us stop if our plan encounters a new obstacle? If so, I think you should rename it since it sounds like this will take us to the previous waypoint.